conda_env/env.yaml

```diff
@@ -1,21 +1,36 @@
 name: rad-gen-env
+#-py-12
 
 channels:
   - conda-forge
 dependencies:
   # ASIC DSE
-  - gdstk=0.9.42
+  - gdstk == 0.9.42
   # - gdspy=1.4 # moved to requirements.txt cus theres no 1.4 on conda channels
-  - matplotlib=3.5.3
-  - pandas=1.4.2
-  - numpy=1.24.1
-  - pyYAML=6.0
-  - python=3.9.13
-  - pandas=1.4.2
-  - pip=23.3.1
+  - matplotlib
+  - pandas
+  - numpy
+  - pyYAML
+  - python==3.12.3
+  - pandas
+  - pip
   # IC 3D 
-  - plotly=5.5.0
+  - plotly
```

Comment:
This is WAY more work than needed, so feel free to ignore this; but since you removed the dependency versions, that means that your code should work for all versions of each of these tools (which is possible if you are only using a SMALL amount of the library; especially if it is a stable part of the library). But if you are using "unstable" features of a library, it may be a good idea to put minimum (or maximum) versions of libraries to prevent issues in the future. For example, in a year, someone may clone your repo and a recent pandas upgrade may cause issues.

Reply:
Note: sometimes its a good idea to lock dependencies on specific version (if they are important) and then upgrading them only when you are certain that specific versions work. Makes upgrades easier and makes the code base a bit more stable since things only change version when you want them to.